plugins/intercom/hooks.js

```diff
@@ -0,0 +1,82 @@
+import { Hooks } from "PluginSDK";
+
+import DOMUtils from "#SRC/js/utils/DOMUtils";
+
+import AuthStore from "#SRC/js/stores/AuthStore";
+import IntercomStore from "#SRC/js/stores/IntercomStore";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+
+import EventTypes from "#SRC/js/constants/EventTypes";
+
+const SDK = require("./SDK").getSDK();
+
+module.exports = {
+  actions: ["pluginsConfigured", "userLoginSuccess", "userLogoutSuccess"],
+
+  initialize(configuration) {
+    this.configuration = configuration;
+
+    this.actions.forEach(action => {
+      SDK.Hooks.addAction(action, this[action].bind(this));
+    });
+
+    this.bootIntercom = this.bootIntercom.bind(this);
+
+    const intercomScript = [
+      `var APP_ID = "${this.configuration.appId}";`,
+      `(function(){var w=window;var ic=w.Intercom;if(typeof ic==="function"){ic('reattach_activator');ic('update',intercomSettings);}else{var d=document;var i=function(){i.c(arguments)};i.q=[];i.c=function(args){i.q.push(args)};w.Intercom=i;function l(){var s=d.createElement('script');s.type='text/javascript';s.async=true;s.src='https://widget.intercom.io/widget/' + APP_ID;var x=d.getElementsByTagName('script')[0];x.parentNode.insertBefore(s,x);}if(w.attachEvent){w.attachEvent('onload',l);}else{w.addEventListener('load',l,false);}}})()`
+    ].join("");
+
+    DOMUtils.appendScript(global.document.head, intercomScript);
```

Comment:
If I get it right this includes a script that loads a bigger script from some backend service? Is it really save to assume that `window.Intercom` is always there? What happens if we block the endpoint or throttle it? I would like is to have a bit more caution here.

Reply:
That might be related with the fact that I do not see the user information on bootIntercom(), on the Intercom Dashboard